internal/util/util.go

```diff
@@ -122,8 +151,27 @@ func CheckDesiredBundleTemplate(existingBundle *rukpakv1alpha1.Bundle, desiredBu
 
 // GenerateBundleName is responsible for generating a unique
 // Bundle metadata.Name using the biName parameter as the base.
-func GenerateBundleName(biName string) string {
-	return fmt.Sprintf("%s-%s", biName, rand.String(5))
+func GenerateBundleName(bi *rukpakv1alpha1.BundleInstance) string {
```

Comment:
This is consistent with the plain provisioner implementation in #377 which is cool. I don't suppose there is any risk of collisions either (between a plain and registry version of a bundle), since the registry bundles would have the registry provisioner ID specified in their template, which would change the generated hash.